internal/pkg/agent/install/progress_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package install
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+type testWriter struct {
+	buf []byte
+}
+
+func newTestWriter() *testWriter {
+	return &testWriter{
+		buf: []byte{},
+	}
+}
+
+func (tw *testWriter) Write(p []byte) (int, error) {
+	tw.buf = append(tw.buf, p...)
+	return len(p), nil
+}
+
+func TestProgress(t *testing.T) {
+	t.Run("single_step_immediate_failure", func(t *testing.T) {
+		w := newTestWriter()
+		pt := NewProgressTracker(w)
+
+		pt.Start()
+
+		pt.StepStart("step 1 starting")
+		pt.StepFailed()
+
+		pt.Stop()
+
+		require.Equal(t, "step 1 starting... FAILED\n", string(w.buf))
```

Comment:
[Blocker]
Isn't there a chance there will be more than 3 dots(`...`) when the assertion happens?
You didn't disable the randomised ticker intervals.

Reply:
We shouldn't see more than three dots because the step starts (line 36) and immediately fails (line 37). Unlike some of the other tests in this PR, there is no delay (sleep) in between those two lines, so more dots will not be shown.